### PR TITLE
Calibrating historical data

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**
Setup for a DAG that calibrates historical data.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-1099](https://airqoteam.atlassian.net/browse/PLAT-1099)

**_HOW DO I TEST OUT THIS PR?_**
Follow instructions in the [Readme file](https://github.com/airqo-platform/AirQo-api/tree/staging/src/airflow#running-using-docker)

### CODE REVIEW

File: src/airflow/airqo_etl_utils/airqo_utils.py [hunk 1]
```diff
@@ -822,8 +822,8 @@ def merge_values(dataframe: pd.DataFrame, variable: str) -> pd.DataFrame:
     return merged_data
 
 
-def calibrate_using_pickle_file(measurements: list) -> list:
-    if not measurements:
+def calibrate_using_pickle_file(measurements: pd.DataFrame) -> list:
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/airqo_utils.py:825]: Using dataframes as a means of data exchange between functions

File: src/airflow/airqo_etl_utils/app_insights_utils.py [hunk 2]
```diff
@@ -312,7 +312,7 @@ def create_insights_data_from_bigquery(
 
     bigquery_api = BigQueryApi()
 
-    hourly_data = bigquery_api.get_hourly_data(
+    hourly_data = bigquery_api.query_data(
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/app_insights_utils.py:315]: renaming function to reduce ambiguity.

File: src/airflow/airqo_etl_utils/bigquery_api.py [hunk 3]
```diff
@@ -19,6 +19,9 @@ def __init__(self):
         self.analytics_table = configuration.BIGQUERY_ANALYTICS_TABLE
         self.sites_table = configuration.BIGQUERY_SITES_TABLE
         self.devices_table = configuration.BIGQUERY_DEVICES_TABLE
+        self.calibrated_hourly_measurements_table = (
+            configuration.BIGQUERY_CALIBRATED_HOURLY_EVENTS_TABLE
+        )
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/bigquery_api.py:24]: table to temporary stored the historical calibrated data for analysis and verification before updating the main tables.

File: src/airflow/airqo_etl_utils/bigquery_api.py [hunk 4]
```diff
@@ -64,6 +68,9 @@ def __get_columns(self, table: str, data_type="") -> list:
         elif table == self.hourly_weather_table or table == self.raw_weather_table:
             schema_path = "schema/weather_data.json"
             schema = "weather_data.json"
+        elif table == self.calibrated_hourly_measurements_table:
+            schema_path = "schema/calibrated_measurements.json"
+            schema = "calibrated_measurements.json"
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/bigquery_api.py:73]: setting schema paths for the temporary table

File: src/airflow/airqo_etl_utils/config.py [hunk 5]
```diff
@@ -24,7 +27,7 @@ class Config:
 
     POST_EVENTS_BODY_SIZE = os.getenv("POST_EVENTS_BODY_SIZE", 10)
     POST_WEATHER_BODY_SIZE = os.getenv("POST_EVENTS_BODY_SIZE", 10)
-    CALIBRATE_REQUEST_BODY_SIZE = os.getenv("CALIBRATE_REQUEST_BODY_SIZE", 10)
+    CALIBRATE_REQUEST_BODY_SIZE = os.getenv("CALIBRATE_REQUEST_BODY_SIZE", 40)
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/config.py:30]: increasing calibrate request body size after performance tuning in #816

File: src/airflow/airqo_etl_utils/schema/calibrated_measurements.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
+[
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/schema/calibrated_measurements.json:1]: schema for tables that will temporarily stop calibrated data during analysis and validation

File: src/airflow/dags/airqo_measurements.py [hunk 7]
```diff
@@ -146,6 +146,144 @@ def load(airqo_data: pd.DataFrame, **kwargs):
...
 
 
+@dag(
+    "AirQo-Historical-Data-Calibration",
+    schedule_interval=None,
+    on_failure_callback=slack_dag_failure_notification,
+    start_date=datetime(2021, 1, 1),
+    catchup=False,
+    tags=["airqo", "calibration", "historical"],
+)
```
Comment (NoahNsimbe) [inline src/airflow/dags/airqo_measurements.py:156]: DAG to calibrate historical data

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**